dragonfly/implementations/at_operator.py

```diff
@@ -0,0 +1,460 @@
+            message = 'The operator listed in Google Calendar is not found in Slack.'
+        elif new_operator_name != self.current_operator_name:
+            self.current_operator_name = new_operator_name
+            self.current_operator_id = self.full_name_to_id_dictionary[self.current_operator_name]
+            self.current_shift_end_time = shift_end_time
+            logger.info(' Found a new operator: ' + new_operator_name)
+            self.send_message(self.monitor_channel_id, "I've found a new operator: " + new_operator_name + ". The shift will end at " + str(shift_end_time))
+        elif shift_end_time > self.current_shift_end_time:
+            self.current_shift_end_time = shift_end_time
+            logger.info(' Extended the shift time for current operator!')
+            message = "It seems that the shift for current operator is extended! \n"
+            message += "I'm not sure whether this branch will be used though."
+        if not regular_check:
+            self.send_message(self.monitor_channel_id, message)
+
+    def construct_user_dictionaries(self):
+        '''
+        Return 3 dictionaries storing information of Slack users: full name to id, id to username, and username to id.
+        '''
+        request = self.slack_client.api_call("users.list")
+        if request['ok']:
+            full_name_to_id_dictionary = {}
+            id_to_username_dictionary = {}
+            username_to_id_dictionary = {}
+            for m in request['members']:
+                if not m['is_bot'] and not m['deleted']:
+                    full_name_to_id_dictionary[m['real_name']] = m['id']
+                    id_to_username_dictionary[m['id']] = m['name']
+                    username_to_id_dictionary[m['name']] = m['id']
+            logger.info(' Constructed dictionaries for full names, Slack usernames and ids.')
+            return full_name_to_id_dictionary, id_to_username_dictionary, username_to_id_dictionary
+        else:
+            return None, None, None
+
+    def construct_channel_dictionaries(self):
+        '''
+        Return 2 dictionaries storing information of Slack channels: channel name to id, and channel id to name.
+        '''
+        logger.info(" Trying to construct an dictionary mapping channel names to their ids.")
+        request = self.slack_client.api_call("conversations.list")
+        if request['ok']:
+            channel_name_to_id_dictionary = {}
+            channel_id_to_name_dictionary = {}
+            for c in request['channels']:
+                channel_name_to_id_dictionary[c['name']] = c['id']
+                channel_id_to_name_dictionary[c['id']] = c['name']
+            logger.info(' Constructed dictionaries containing information of ' + str(len(request['channels'])) + ' channels.')
+            return channel_name_to_id_dictionary, channel_id_to_name_dictionary
```

Comment:
when I tried accessing the dictionaries they were blank, I'm unsure why given that the log said `len(request['channels'])` == 73

Reply:
I tried to access the dict and it gave me a really long one - and it is supposed to contain all the channels in the workspalce :((( Maybe just try again...?

Reply:
maybe something went sideways in startup when I was testing it
cannot reproduce is good with me, let's mark this resolved